src/routeGeneration/routeGenerator.ts

```diff
@@ -31,7 +31,7 @@ export class RouteGenerator {
   constructor(private readonly metadata: Tsoa.Metadata, private readonly options: RoutesConfig, private readonly minimalSwaggerConfig: SwaggerConfigRelatedToRoutes) { }
 
   public async GenerateRoutes(middlewareTemplate: string, pathTransformer: (path: string) => string) {
-    const fileName = `${this.options.routesDir}/routes.ts`;
+    const fileName = `${this.options.routesDir}/${this.options.routesFileName}`;
```

Comment:
Don't set the default in `cli.ts` because there are users that use tsoa programmatically that would not get the default value. Please move it here.

So basically:
```ts
const routesFileNameDefault = `routes.ts`;
const routesFileName = this.options.routesFileName ? this.options.routesFileName : routesFileNameDefault;
const fileName = `${this.options.routesDir}/${routesFileName}`;
```

Reply:
I knew I was doing something wrong. It just felt wrong every step of the way... I'll get to changing it.